packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,252 @@
+   */
+  readonly provisioningArtifacts: ProvisioningArtifactProperties[];
+
+  /**
+   * ReplaceProvisioningArtifacts
+   * @default false
+   */
+  readonly replaceProvisioningParameters?: boolean;
+
+  /**
+   * The support information about the product
+   * @default - No support description provided
+   */
+  readonly supportDescription?: string;
+
+  /**
+   * The contact email for product support.
+   * @default - No support email provided
+   */
+  readonly supportEmail?: string;
+
+  /**
+   * The contact URL for product support.
+   * @default - No support URL provided
+   */
+  readonly supportUrl?: string;
+
+  /**
+   * One or more tags.
+   * @default - No tags provided
+   */
+  readonly tags?: Tag[];
+}
+
+/**
+ * A Service Catalog Cloudformation Product.
+ * @resource AWS::ServiceCatalog::CloudFormationProduct
+ */
+export class Product extends ProductBase {
+
+  /**
+   * Creates a Product construct that represents an external product.
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs Product import properties
+   */
+  public static fromProductAttributes(scope: Construct, id: string, attrs: ProductAttributes): IProduct {
+    const parts = Stack.of(scope).parseArn(attrs.productArn);
+    function throwError(errorMessage: string): never {
+      throw new Error(errorMessage);
+    }
+    class Import extends ProductBase {
+      public readonly productId = parts.resourceName ?? throwError('Product arn missing Product ID during import from attributes');
+      public readonly productArn = attrs.productArn;
+      public readonly productName = attrs.productName;
+    }
+    return new Import(scope, id);
+  }
+
+  public readonly productArn: string;
+  public readonly productName: string;
+  public readonly productId: string;
+  private readonly product: CfnCloudFormationProduct;
+
+  constructor(scope: Construct, id: string, props: ProductProps) {
+    super(scope, id, {
+      physicalName: props.productName,
+    },
+    );
+
+    this.product = new CfnCloudFormationProduct(this, 'Resource', {
+      acceptLanguage: props.acceptLanguage,
+      description: props.description,
+      distributor: props.distributor,
+      name: props.productName,
+      owner: props.owner,
+      provisioningArtifactParameters: this.generateProvisioningArtifactParameters(props),
+      replaceProvisioningArtifacts: props.replaceProvisioningParameters,
+      supportDescription: props.supportDescription,
+      supportEmail: props.supportEmail,
+      supportUrl: props.supportUrl,
+      tags: props.tags,
+    });
+
+    this.productArn = this.getResourceArnAttribute(this.product.ref, {
+      service: 'catalog',
+      resource: 'product',
+      resourceName: this.product.ref,
+    });
+
+    this.productName = this.getResourceNameAttribute(props.productName);
+    this.productId = this.getResourceNameAttribute(this.product.ref);
+  }
```

Comment:
Same question here: is `productName` not part of the ARN, but instead a generated `productId` is?

Reply:
correct, the ARNs look like `arn:aws:catalog:us-east-1:012345678990:portfolio/port-c5e24l5fou7x6` for products it has `port-`

Reply:
> correct, the ARNs look like `arn:aws:catalog:us-east-1:012345678990:portfolio/port-c5e24l5fou7x6`

So, just to be 100% sure I understand, the Portfolio ID is `port-c5e24l5fou7x`? And it's an automatically-generated value at creation time, that cannot be set by the customer?

Reply:
correct